src/formats/ktx2.rs

```diff
@@ -0,0 +1,32 @@
+use std::io::{self, BufRead, Seek, SeekFrom};
+
+use crate::{ImageResult, ImageSize};
+
+pub fn size<R: BufRead + Seek>(reader: &mut R) -> ImageResult<ImageSize> {
+    reader.seek(SeekFrom::Start(0))?;
+
+    let mut identifier = [0; 12];
+    reader.read_exact(&mut identifier)?;
+    let ktx2_identifier = [
+        0xAB, 0x4B, 0x54, 0x58, 0x20, 0x32, 0x30, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A,
+    ];
+
+    if identifier != ktx2_identifier {
+        return Err(io::Error::new(io::ErrorKind::InvalidData, "Invalid KTX2 identifier").into());
+    }
+
+    let mut header = [0; 40];
```

Comment:
Same as DDS, no need to read the entire header. Just seek to the offset of width/height and use crate util methods to read the integers.